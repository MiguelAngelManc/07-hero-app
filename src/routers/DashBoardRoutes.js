import React from 'react'
import { Redirect, Route, Switch } from 'react-router-dom/cjs/react-router-dom.min'
import { DcScreen } from '../Components/dc/DcScreen'
import { HeroScreen } from '../Components/heroes/HeroScreen'
import { MarvelScreen } from '../Components/marvel/MarvelScreen'
import { SearchScreen } from '../Components/search/SearchScreen'
import { Navbar } from '../Components/ui/Navbar'

export const DashBoardRoutes = () => {
    return (
        <>
            <Navbar />

            <div className="container">

                <Switch>

                    <Route exact path="/marvel" component={ MarvelScreen }/>
                    <Route exact path="/hero/:heroeId" component={ HeroScreen }/>
                    <Route exact path="/dc" component={ DcScreen }/>
                    <Route path="/search" component={ SearchScreen }/>

                    <Redirect to="/marvel"/>
                    
                </Switch>

            </div>
        </>
    )
}
