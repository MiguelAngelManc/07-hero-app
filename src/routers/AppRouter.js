import React from "react";
import {
    BrowserRouter as Router,
    Switch,
    Route,
    Link
} from "react-router-dom";
import { DcScreen } from "../Components/dc/DcScreen";
import { LoginScreen } from "../Components/login/LoginScreen";
import { MarvelScreen } from "../Components/marvel/MarvelScreen";
import { Navbar } from "../Components/ui/Navbar";
import { DashBoardRoutes } from "./DashBoardRoutes";

export const AppRouter = () => {
    return (
        <Router>
            <div>

                <Switch>
                    <Route exact path="/login" component={ LoginScreen } />
                    
                    <Route path="/" component={ DashBoardRoutes } />
                    

                </Switch>
            </div>
        </Router>
    );
}
    
