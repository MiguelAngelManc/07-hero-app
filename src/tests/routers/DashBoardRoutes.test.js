import { mount } from "enzyme"
import { MemoryRouter } from "react-router-dom"
import { AuthContext } from "../../auth/AuthContext"
import { DashBoardRoutes } from "../../routers/DashBoardRoutes"


describe('Pruebas con DashBoardRoutes', () => {
    
    const contextValue = {
        dispatch:jest.fn(),
        user:{
            logged:true,
            name:'Miguel'
        }
    }

    test('Match snapshot', () => {
        
        const wrapper = mount(
            <AuthContext.Provider value={contextValue}>
                <MemoryRouter>
                    <DashBoardRoutes />
                </MemoryRouter>
            </AuthContext.Provider>
        )

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('.text-info').text().trim()).toBe('Miguel')
    })
    

})
