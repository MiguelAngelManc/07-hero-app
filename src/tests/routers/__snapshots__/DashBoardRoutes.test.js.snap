// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas con DashBoardRoutes Match snapshot 1`] = `
Array [
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      Asociaciones
    </a>
    <div
      className="navbar-collapse"
    >
      <div
        className="navbar-nav"
      >
        <a
          aria-current="page"
          className="nav-item nav-link active"
          href="/marvel"
          onClick={[Function]}
          style={Object {}}
        >
          Marvel
        </a>
        <a
          aria-current={null}
          className="nav-item nav-link"
          href="/dc"
          onClick={[Function]}
        >
          DC
        </a>
        <a
          aria-current={null}
          className="nav-item nav-link"
          href="/search"
          onClick={[Function]}
        >
          Search...
        </a>
      </div>
    </div>
    <div
      className="navbar-collapse collapse w-100 order-3 dual-collapse2"
    >
      <ul
        className="navbar-nav ml-auto"
      >
        <p
          className="nav-item nav-link text-info"
        >
          Miguel
        </p>
        <button
          activeClassName="active"
          className="nav-item nav-link btn"
          exact={true}
          onClick={[Function]}
          to="/login"
        >
          Logout
        </button>
      </ul>
    </div>
  </nav>,
  <div
    className="container"
  >
    Array [
      <h1>
        Marvel screen
      </h1>,
      <hr />,
      <div
        className="row row-cols-1 animate__animated animate__fadeIn"
      >
        <div
          className="card ms-3"
          style={
            Object {
              "maxWidth": 550,
            }
          }
        >
          <div
            className="row no-gutters"
          >
            <div
              className="column col-md-4"
            >
              <img
                alt="Heroe"
                className="card-img"
                src="./assets/heroes/marvel-spider.jpg"
              />
              <div
                className="col-md-8"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Spider Man
                  </h5>
                  <p
                    className="card-text"
                  >
                    Peter Parker
                  </p>
                  <p
                    className="card-text"
                  >
                    <small
                      className="text-muted"
                    >
                      Amazing Fantasy #15
                    </small>
                  </p>
                  <a
                    href="/hero/marvel-spider"
                    onClick={[Function]}
                  >
                    Mas...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card ms-3"
          style={
            Object {
              "maxWidth": 550,
            }
          }
        >
          <div
            className="row no-gutters"
          >
            <div
              className="column col-md-4"
            >
              <img
                alt="Heroe"
                className="card-img"
                src="./assets/heroes/marvel-captain.jpg"
              />
              <div
                className="col-md-8"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Captain America
                  </h5>
                  <p
                    className="card-text"
                  >
                    Steve Rogers
                  </p>
                  <p
                    className="card-text"
                  >
                    <small
                      className="text-muted"
                    >
                      Captain America Comics #1
                    </small>
                  </p>
                  <a
                    href="/hero/marvel-captain"
                    onClick={[Function]}
                  >
                    Mas...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card ms-3"
          style={
            Object {
              "maxWidth": 550,
            }
          }
        >
          <div
            className="row no-gutters"
          >
            <div
              className="column col-md-4"
            >
              <img
                alt="Heroe"
                className="card-img"
                src="./assets/heroes/marvel-iron.jpg"
              />
              <div
                className="col-md-8"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Iron Man
                  </h5>
                  <p
                    className="card-text"
                  >
                    Tony Stark
                  </p>
                  <p
                    className="card-text"
                  >
                    <small
                      className="text-muted"
                    >
                      Tales of Suspense #39
                    </small>
                  </p>
                  <a
                    href="/hero/marvel-iron"
                    onClick={[Function]}
                  >
                    Mas...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card ms-3"
          style={
            Object {
              "maxWidth": 550,
            }
          }
        >
          <div
            className="row no-gutters"
          >
            <div
              className="column col-md-4"
            >
              <img
                alt="Heroe"
                className="card-img"
                src="./assets/heroes/marvel-thor.jpg"
              />
              <div
                className="col-md-8"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Thor
                  </h5>
                  <p
                    className="card-text"
                  >
                    Thor Odinson
                  </p>
                  <p
                    className="card-text"
                  >
                    <small
                      className="text-muted"
                    >
                      Journey into Myster #83
                    </small>
                  </p>
                  <a
                    href="/hero/marvel-thor"
                    onClick={[Function]}
                  >
                    Mas...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card ms-3"
          style={
            Object {
              "maxWidth": 550,
            }
          }
        >
          <div
            className="row no-gutters"
          >
            <div
              className="column col-md-4"
            >
              <img
                alt="Heroe"
                className="card-img"
                src="./assets/heroes/marvel-hulk.jpg"
              />
              <div
                className="col-md-8"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Hulk
                  </h5>
                  <p
                    className="card-text"
                  >
                    Bruce Banner
                  </p>
                  <p
                    className="card-text"
                  >
                    <small
                      className="text-muted"
                    >
                      The Incredible Hulk #1
                    </small>
                  </p>
                  <a
                    href="/hero/marvel-hulk"
                    onClick={[Function]}
                  >
                    Mas...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card ms-3"
          style={
            Object {
              "maxWidth": 550,
            }
          }
        >
          <div
            className="row no-gutters"
          >
            <div
              className="column col-md-4"
            >
              <img
                alt="Heroe"
                className="card-img"
                src="./assets/heroes/marvel-wolverine.jpg"
              />
              <div
                className="col-md-8"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Wolverine
                  </h5>
                  <p
                    className="card-text"
                  >
                    James Howlett
                  </p>
                  <p
                    className="card-text"
                  >
                    <small
                      className="text-muted"
                    >
                      The Incredible Hulk #180
                    </small>
                  </p>
                  <a
                    href="/hero/marvel-wolverine"
                    onClick={[Function]}
                  >
                    Mas...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card ms-3"
          style={
            Object {
              "maxWidth": 550,
            }
          }
        >
          <div
            className="row no-gutters"
          >
            <div
              className="column col-md-4"
            >
              <img
                alt="Heroe"
                className="card-img"
                src="./assets/heroes/marvel-daredevil.jpg"
              />
              <div
                className="col-md-8"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Daredevil
                  </h5>
                  <p
                    className="card-text"
                  >
                    Matthew Michael Murdock
                  </p>
                  <p
                    className="card-text"
                  >
                    <small
                      className="text-muted"
                    >
                      Daredevil #1
                    </small>
                  </p>
                  <a
                    href="/hero/marvel-daredevil"
                    onClick={[Function]}
                  >
                    Mas...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card ms-3"
          style={
            Object {
              "maxWidth": 550,
            }
          }
        >
          <div
            className="row no-gutters"
          >
            <div
              className="column col-md-4"
            >
              <img
                alt="Heroe"
                className="card-img"
                src="./assets/heroes/marvel-hawkeye.jpg"
              />
              <div
                className="col-md-8"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Hawkeye
                  </h5>
                  <p
                    className="card-text"
                  >
                    Clinton Francis Barton
                  </p>
                  <p
                    className="card-text"
                  >
                    <small
                      className="text-muted"
                    >
                      Tales of Suspense #57
                    </small>
                  </p>
                  <a
                    href="/hero/marvel-hawkeye"
                    onClick={[Function]}
                  >
                    Mas...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card ms-3"
          style={
            Object {
              "maxWidth": 550,
            }
          }
        >
          <div
            className="row no-gutters"
          >
            <div
              className="column col-md-4"
            >
              <img
                alt="Heroe"
                className="card-img"
                src="./assets/heroes/marvel-cyclops.jpg"
              />
              <div
                className="col-md-8"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Cyclops
                  </h5>
                  <p
                    className="card-text"
                  >
                    Scott Summers
                  </p>
                  <p
                    className="card-text"
                  >
                    <small
                      className="text-muted"
                    >
                      X-Men #1
                    </small>
                  </p>
                  <a
                    href="/hero/marvel-cyclops"
                    onClick={[Function]}
                  >
                    Mas...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card ms-3"
          style={
            Object {
              "maxWidth": 550,
            }
          }
        >
          <div
            className="row no-gutters"
          >
            <div
              className="column col-md-4"
            >
              <img
                alt="Heroe"
                className="card-img"
                src="./assets/heroes/marvel-silver.jpg"
              />
              <div
                className="col-md-8"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Silver Surfer
                  </h5>
                  <p
                    className="card-text"
                  >
                    Norrin Radd
                  </p>
                  <p
                    className="card-text"
                  >
                    <small
                      className="text-muted"
                    >
                      The Fantastic Four #48
                    </small>
                  </p>
                  <a
                    href="/hero/marvel-silver"
                    onClick={[Function]}
                  >
                    Mas...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>,
    ]
  </div>,
]
`;
